runtime/src/main/cpp/KAssert.h

```diff
@@ -46,24 +53,30 @@ extern "C" const int KonanNeedDebugInfo;
 
 #if KONAN_ENABLE_ASSERT
 // Use RuntimeAssert() in internal state checks, which could be ignored in production.
-#define RuntimeAssert(condition, format, ...)                                     \
-if (KonanNeedDebugInfo && (!(condition))) {                                       \
-    RuntimeAssertFailed( __FILE__ ":" TOSTRING(__LINE__), format, ##__VA_ARGS__); \
-}
+#define RuntimeAssert(condition, format, ...) \
+    do { \
+        if (KonanNeedDebugInfo && (!(condition))) { \
+            RuntimeAssertFailed(CURRENT_SOURCE_LOCATION, format, ##__VA_ARGS__); \
+        } \
+    } while (false)
 #else
-#define RuntimeAssert(condition, message)
+#define RuntimeAssert(condition, format, ...) \
+    do { \
+    } while (false)
 #endif
 
 // Use RuntimeCheck() in runtime checks that could fail due to external condition and shall lead
 // to program termination. Never compiled out.
-#define RuntimeCheck(condition, format, ...)             \
-  if (!(condition)) {                                    \
-    RuntimeAssertFailed(nullptr, format, ##__VA_ARGS__); \
-  }
+#define RuntimeCheck(condition, format, ...) \
+    do { \
+        if (!(condition)) { \
+            RuntimeAssertFailed(CURRENT_SOURCE_LOCATION, format, ##__VA_ARGS__); \
```

Comment:
So this enables source location info for `RuntimeCheck` even in release binaries, thus adds more strings to the binaries. Is this intended?

Reply:
My bad, fixed.